components/wifi/esp8266-driver/ESP8266Interface.cpp

```diff
@@ -479,6 +481,11 @@ int ESP8266Interface::socket_send(void *handle, const void *data, unsigned size)
 
     status = _esp.send(socket->id, data, size);
 
+    if (status == NSAPI_ERROR_WOULD_BLOCK)
```

Comment:
If `event` is the thing that leads to sigio, this should really be timed, or you're just immediately re-entering, probably. I would expect this to be using "call_in" for the `event`, whereas the other timeout stuff in the PR could just be checking elapsed time in the send call, I think. (although I'm not 100% sure what that's doing)

Reply:
the status == NSAPI_ERROR_WOULD_BLOCK will only be returned from ESP8266::send function if both _busy and _busy_timeout_reached are set. 
To set the _busy_timeout_reached flag we need to call the oob_busy_timeout() function which is only called in ESP8266Interface::_oob_busy_timeout() and that is called via a call_in from _oob_busy_detected().
So indirectly - this is timed. If the send returns with a valid return error within 1/10s the fact that there was a busy flag will be ignored.
Does this make sense?

Reply:
Um, not sure. Seems too complicated, and not sure there's a need.

If the device does promptly return BUSY or OK after each command, then can just finish on either of those. Timeout shouldn't matter if it's working.

After you get BUSY, check overall elapsed time to determine whether you want to try again, or think it's time to go to WOULDBLOCK. Just look at `Kernel::get_ms_count()` to note time - don't need a callback, or any special handling in the busy OOB (just set busy flag and abort wait for OK).